sw/device/lib/testing/test_framework/ottf_common.ld

```diff
@@ -274,28 +272,39 @@ SECTIONS {
    * twice, tests that require non-volatile counters can use this section by
    * striking flash words to count up to `_non_volatile_counter_max_count`.
    */
-  .non_volatile_counter_0 _non_volatile_counter_0_start (NOLOAD) : ALIGN(2048) {
```

Comment:
Maybe add a comment here that those sections were reordered so that the order corresponds to memory order?